### PR TITLE
Addition of Project Section

### PR DESCRIPTION
**Pull request by Arioum**
### The Project section consists of following features:
- In the main homepage, it has a dedicated section where each projects are displayed in the form carousel of cards with minimal info.
- If more info is needed, a link is provided to take you to its dedicated project details.

### Implementation:
- Used File Routing API of Gatsby to achieve programmatic page creation whenever a new project is created in the Decap CMS.
- Created a project template page which will act as the template for any incoming project data.

image

### CODE REVIEW

**Comment by Arioum** [comment]
### [FIXED]KNOWN ISSUE: 
- Issue with the individual project page being blank when clicked on `View details` on the home page. (Requires Clearing cache of netlify build step)

**Review by shivah12 — COMMENTED** [review]
I don't see any issue with this PR

File: src/components/Projects.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  return (
+    <Flex gap={4} justifyContent={'center'} mt={4}>
+      <IconButton
+        onClick={() => {
+          swiper.slidePrev();
+          handleButtonAction('left');
+        }}
+        disabled={currentIndex === 0}
+        p='0'
```
Comment (saurabh73) [inline src/components/Projects.tsx:47]: is this not default padding value?

File: src/components/Projects.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    <WrapItem
+      key={index}
+      justifyContent='space-between'
+      flexDirection='column'
+      color='black.300'
+      bg='base.d100'
+      rounded={5}
+      flex={1}
+      py={5}
+      pt={0}
```
Comment (saurabh73) [inline src/components/Projects.tsx]: is this not default padding value?

File: src/components/Projects.tsx (hunk 1 above)
Comment (saurabh73) [inline src/components/Projects.tsx]: is this not default padding value?

File: src/components/Projects.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+type NodeType = {
+  frontmatter: {
+    title: string | null;
+    description: string | null;
+    image: string | null;
+    slug: string | null;
+  } | null;
+};
+
+const Slide = ({ node, index }: { node: NodeType; index: number }) => {
```
Comment (saurabh73) [inline src/components/Projects.tsx]: This should be in Separate Component File, with name `ProjectSlide`

File: src/components/Projects.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+const SwiperNavButtons: React.FC<SwiperNavButtonsProps> = ({
+  handleButtonAction,
+  currentIndex,
+  maxIndex,
+}) => {
+  const swiper = useSwiper();
+
+  return (
+    <Flex gap={4} justifyContent={'center'} mt={4}>
+      <IconButton
```
Comment (saurabh73) [inline src/components/Projects.tsx:25]: Icon Color should have a `dark` mode bg.

File: src/components/Projects.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        disabled={currentIndex >= maxIndex}
+        p='0'
+        size='fit-content'
+        backgroundColor='transparent'
+        _hover={{ backgroundColor: 'transparent' }}
+        aria-label={'move right'}
+        icon={
+          <FaCircleChevronRight
+            size={30}
+            color={currentIndex >= maxIndex ? 'gray' : 'black'}
```
Comment (saurabh73) [inline src/components/Projects.tsx]: Color should be coming from current theme, light or dark

File: src/components/Projects.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+          justifyContent={'center'}
+          alignItems={'center'}
+        >
+          <Image
+            src={node.frontmatter?.image || undefined}
+            alt={node.frontmatter?.title || undefined}
+            rounded={3}
+            w={'full'}
+            minWidth={'fit-content'}
+            h={'full'}
```
Comment (saurabh73) [inline src/components/Projects.tsx]: add `objectFit` property also.

File: src/components/Projects.tsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+      setCurrentIndex((prevIndex) => Math.min(prevIndex + 1, maxIndex));
+    }
+    if (action === 'left') {
+      setCurrentIndex((prevIndex) => Math.max(prevIndex - 1, 0));
+    }
+  };
+
+  const [isLargerThan968] = useMediaQuery('(min-width: 968px)');
+  const [isLargerThan768] = useMediaQuery('(min-width: 768px)');
+  const cardsToShow = (isLargerThan968 && 3) || (isLargerThan768 && 2) || 1;
```
Comment (saurabh73) [inline src/components/Projects.tsx]: This is swiper settings. https://swiperjs.com/swiper-api#param-breakpoints

File: src/views/ProjectTemplate.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    return <div>Loading...</div>;
+  }
+
+  const { markdownRemark } = data;
+  const { frontmatter, html } = markdownRemark;
+  const imagePath = frontmatter.image;
+
+  return (
+    <>
+      <Navbar />
```
Comment (saurabh73) [inline src/views/ProjectTemplate.tsx]: Use Layout for common page elements: https://www.gatsbyjs.com/docs/how-to/routing/layout-components/#what-are-layout-components

File: static/admin/config.yml [hunk 9]
```diff
@@ -47,6 +47,24 @@ collections:
...
+      - {
+          label: "Template Key",
+          name: "templateKey",
+          widget: "hidden",
+          default: "project-page",
+        }
+      - { label: "Project Title", name: "title", widget: "string" }
+      - { label: "Project Slug", name: "slug", widget: "string" }
+      - { label: "Image", name: "image", widget: "image"}
+      - { label: "Description", name: "description", widget: "string" }
```
Comment (saurabh73) [inline static/admin/config.yml]: description widget type should be `text`

**Review by saurabh73 — CHANGES_REQUESTED** [review]

File: src/components/Projects.tsx (hunk 1 above)
Comment (Arioum) [reply src/components/Projects.tsx]: Yes, I removed it

File: src/components/Projects.tsx (hunk 4 above)
Comment (Arioum) [reply src/components/Projects.tsx:25]: icon does not have a `bg` in both scenario. The default `bg` is set to transparent as it is not required in both dark and light mode.

File: src/components/Projects.tsx (hunk 7 above)
Comment (Arioum) [reply src/components/Projects.tsx]: I just changed the breakpoint settings to utilize the inbuilt swiper breakpoints. However I wont be able to remove the above `useMediaQuery` as it is required to determine the index of the slides to properly indicate whether if its is the first or the last slide. This information is used to control the slide button status i.e `disabled` when at the end or starting slide.

Comment (saurabh73) [reply src/components/Projects.tsx]: @Arioum that is also not needed,  you can use something like this, and pass these as props to `SwiperNavButtons`

```
 const [isBeginning, setIsBeginning] = useState<boolean>(true)
  const [isEnd, setIsEnd] = useState<boolean>(false)
<Swiper
        ...
         onActiveIndexChange={(swiper) => {
                  setIsBeginning(swiper.isBeginning)
                  setIsEnd(swiper.isEnd)
         }}
          ...
         >
 ```

File: src/components/Projects.tsx [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+        _hover={{ backgroundColor: 'transparent' }}
+        aria-label={'move left'}
+        icon={<FaCircleChevronLeft size={30} />}
+      />
+      <IconButton
+        onClick={() => {
+          swiper.slideNext();
+          handleButtonAction('right');
+        }}
+        isDisabled={currentIndex >= maxIndex}
```
Comment (saurabh73) [inline src/components/Projects.tsx]: pass this as prop, `isBeginning` or `isEnd`

**Review by saurabh73 — CHANGES_REQUESTED** [review]

File: src/components/layout.tsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  children: ReactNode;
+}
+
+const Layout: React.FC<LayoutProps> = ({ children }) => {
+  return (
+    <>
+      <Navbar />
+      <Container maxW={'6xl'} as='main' my={4}>
+        {children}
+      </Container>
```
Comment (saurabh73) [inline src/components/layout.tsx]: This needs footer as well

**Review by saurabh73 — CHANGES_REQUESTED** [review]

**Review by saurabh73 — APPROVED** [review]

**Pull request merged**